backend/plugin/parser/sql/snowsql.go

```diff
@@ -0,0 +1,41 @@
+// Package parser is the parser for SQL statement.
+package parser
+
+import (
+	"strings"
+
+	"github.com/antlr4-go/antlr/v4"
+	parser "github.com/bytebase/snowsql-parser"
+)
+
+// ParseSnowSQL parses the given SQL statement by using antlr4. Returns the AST and token stream if no error.
+func ParseSnowSQL(statement string) (antlr.Tree, error) {
+	statement = strings.TrimRight(statement, " \t\n\r\f;") + ";"
```

Comment:
```suggestion
	statement = strings.TrimRight(statement, " \t\n\r\f;") + "\n;"
```
Consider this case:
SELECT * FROM t; -- this is comments